hwmgr-plugins/metal3/controller/baremetalhost_manager.go

```diff
@@ -345,6 +345,11 @@ func processHwProfile(ctx context.Context,
 	pluginNamespace string,
 	bmh *metal3v1alpha1.BareMetalHost, profileName string, postInstall bool) (bool, error) {
 
+	// Clear any existing update annotations to ensure clean state
```

Comment:
Possible this is called if a update is already in progress ?

Reply:
Yes, this is for retry scenarios: if a previous request times out while an update is already in progress, the annotation should be cleared when that timeout is detected.